tests/tests.rs

```diff
@@ -597,3 +597,31 @@ fn test_builder_type() {
     let builder: __FooBuilder<_> = Foo::builder();
     assert!(builder.x(1).build() == Foo { x: 1 });
 }
+
+#[test]
+fn test_default_builder_type() {
+    #[derive(Debug, PartialEq, TypedBuilder)]
+    #[builder(builder_type(name = InnerBuilder))]
+    struct Inner {
+        a: i32,
+        b: i32,
+    }
+
+    impl Inner {
+        pub fn outer(self) -> Outer {
+            Outer(self)
+        }
+    }
+
+    #[derive(Debug, PartialEq)]
+    struct Outer(Inner);
+
+    impl Outer {
+        pub fn builder() -> InnerBuilder {
+            Inner::builder()
+        }
+    }
+
+    let outer = Outer::builder().a(3).b(5).build().outer();
```

Comment:
I really hate how `Outher::builder()` does not build an `outher`, but since it's not part of the actual API I'll allow it.

Reply:
Yes, me too. I would really like it to return outer. But that needs some kind of #52, e.g. a possibility to set another type the builder outputs by using `.into()`.

I would really love something like this:
```rust
#[derive(TypedBuilder)]
#[builder(build_method(output = Outer))]
struct Inner { ... }
```
where `Outer` implements `From<Inner>` or something.. It is quite difficult to make your own build method or customizing it, so I don't really see a better way to allow such a pattern.